test/common/network/proxy_protocol_test.cc

```diff
@@ -53,7 +54,11 @@ class ProxyProtocolTest : public testing::Test {
   std::shared_ptr<MockReadFilter> read_filter_;
 };
 
-TEST_F(ProxyProtocolTest, Basic) {
+// Parameterize the listener socket address version.
+INSTANTIATE_TEST_CASE_P(IpVersions, ProxyProtocolTest,
+                        testing::ValuesIn(TestEnvironment::getIpVersionsForTest()));
```

Comment:
The listener, client connection, and writes would need to be mocked. I think it may be simplier just to parameterize the test.